x/incentives/keeper/gauge.go

```diff
@@ -295,69 +295,6 @@ func (k Keeper) getWeightBySplittingPolicy(ctx sdk.Context, poolId uint64, split
 	}
 }
 
-// AddToGaugeRewardsFromGauge transfer coins from a group gauge to its internal gauges.
-// Prior to calling this function, we make sure that the internalGaugeId is linked with the associated groupGaugeId.
-// Note: we donot have to bankSend for this gauge transfer because all the available incentive has already been bank sent
-// when we create Group Gauge. Now we are just allocating funds from groupGauge to internalGauge.
-// CONTRACT: internalGaugeId should be present in the group gauge.
-// TODO: revisit, clean up, and test this function as part of https://github.com/osmosis-labs/osmosis/issues/6402
-func (k Keeper) AddToGaugeRewardsFromGauge(ctx sdk.Context, groupGaugeId uint64, coins sdk.Coins, internalGaugeId uint64) error {
-	groupGauge, err := k.GetGaugeByID(ctx, groupGaugeId)
-	if err != nil {
-		return err
-	}
-
-	internalGauge, err := k.GetGaugeByID(ctx, internalGaugeId)
-	if err != nil {
-		return err
-	}
-
-	if internalGauge.IsFinishedGauge(ctx.BlockTime()) {
-		return errors.New("gauge is already completed")
-	}
-
-	// check if there is sufficient funds in groupGauge to make the transfer
-	remainingCoins := groupGauge.Coins.Sub(groupGauge.DistributedCoins)
-	if remainingCoins.IsAllLT(coins) {
-		return fmt.Errorf("group gauge id: %d doesnot have enough tokens to transfer", groupGaugeId)
-	}
-
-	internalGauge.Coins = internalGauge.Coins.Add(coins...)
-	err = k.setGauge(ctx, internalGauge)
-	if err != nil {
-		return err
-	}
-
-	return nil
-}
-
-// AddToGaugeRewards adds coins to gauge.
-func (k Keeper) AddToGaugeRewards(ctx sdk.Context, owner sdk.AccAddress, coins sdk.Coins, gaugeID uint64) error {
```

Comment:
Note to reviewer: I moved this method below because the original diff was too confusing.

The actual changes are minimal - simply extracting an `addToGaugeRewards` helper